diff --git a/libretro-common/streams/file_stream_transforms.c b/libretro-common/streams/file_stream_transforms.c
index e94a3112..d3664dd4 100644
--- a/libretro-common/streams/file_stream_transforms.c
+++ b/libretro-common/streams/file_stream_transforms.c
@@ -103,7 +103,8 @@ int64_t rfseek(RFILE* stream, int64_t offset, int origin)
          break;
    }
 
-   return filestream_seek(stream, offset, seek_position);
+   /* Must enforce fseek behavior instead of lseek */
+   return filestream_seek(stream, offset, seek_position) < 0 ? -1 : 0;
 }
 
 int64_t rfread(void* buffer,
